### PR TITLE
Fixed Bug 555

### PR DESCRIPTION
**Pull request by bgsavage**
This broke because the translate was now taking longer to resolve now that we are merging ui / api translations.  When the primeng dropdown was being initialized, a placeholder (from translate) hadn't finished resolving so the primeng was selecting the first item in the dropdown rather than using it's placeholder.  Primeng never updated itself once the translation completed.

So now optimized the translate loader by parallelizing the api / ui translation requests and added a translate resolve so that the HomeComponent won't load until translate is complete.

### CODE REVIEW

File: webapp/src/main/webapp/src/app/app.component.ts [hunk 1]
```diff
@@ -9,12 +10,13 @@ import { UserService } from "./user/user.service";
...
 
   private _user;
+  public loadingTranslations: boolean = true;
 
   get user() {
     return this._user;
   }
 
-  constructor(public translate: TranslateService, private _userService: UserService) {
+  constructor(public translate: TranslateService, private _userService: UserService, private translateLoader: RdwTranslateLoader) {
```
Comment (bgsavage) [inline webapp/src/main/webapp/src/app/app.component.ts]: oops these changes can be discarded, was trying to something else for a while.

File: webapp/src/main/webapp/src/app/shared/common.module.ts [hunk 2]
```diff
@@ -41,8 +37,7 @@ export function createTranslateLoader(http: Http) {
     TranslateModule.forRoot({
       loader: {
         provide: TranslateLoader,
-        useFactory: createTranslateLoader,
-        deps: [ Http ]
+        useClass: RdwTranslateLoader
```
Comment (bgsavage) [inline webapp/src/main/webapp/src/app/shared/common.module.ts]: unrelated refactor: now useClass instead of a factory.

File: webapp/src/main/webapp/src/app/shared/rdw-translate-loader.ts [hunk 3]
```diff
@@ -1,25 +1,42 @@
...
-          return Observable.of(merged);
-        }));
+    let uiObservable = this.uiLoader.getTranslation(lang);
+    let apiObservable = this.apiLoader.getTranslation(lang);
+
+    let translateObserver: Observer<any>;
+    this._observable = new Observable<any>(observer => translateObserver = observer).share();
+
+    Observable
+      .forkJoin([uiObservable, apiObservable])
```
Comment (bgsavage) [inline webapp/src/main/webapp/src/app/shared/rdw-translate-loader.ts]: forkJoin is the new $q.all

File: webapp/src/main/webapp/src/app/shared/common.module.ts (hunk 2 above)
Comment (esotrope) [inline webapp/src/main/webapp/src/app/shared/common.module.ts]: Oh the factory here was originally to support some compile time optimization angular cli warned about. You don't get a warning with this change?

File: webapp/src/main/webapp/src/app/shared/rdw-translate-loader.ts (hunk 3 above)
Comment (esotrope) [inline webapp/src/main/webapp/src/app/shared/rdw-translate-loader.ts]: Is spoonJoin the new Promise.all?

Comment (bgsavage) [inline webapp/src/main/webapp/src/app/shared/rdw-translate-loader.ts]: I think it's dinner time for someone :)

Comment (esotrope) [inline webapp/src/main/webapp/src/app/shared/rdw-translate-loader.ts]: 🍽  😋

File: webapp/src/main/webapp/src/app/shared/common.module.ts (hunk 2 above)
Comment (bgsavage) [inline webapp/src/main/webapp/src/app/shared/common.module.ts]: no compile warning here
